Drop use of codecs.open()

s-t-e-v-e-n-k opened the pull request:
codecs.open() has been deprecated as of Python 3.14, and plain open() supports all use-cases we have, switch to using it.

s-t-e-v-e-n-k committed:
chore: Drop use of codecs.open()

codecs.open() has been deprecated as of Python 3.14, and plain open()
supports all use-cases we have, switch to using it.

[pre-commit.ci] auto fixes from pre-commit.com hooks: for more information, see https://pre-commit.ci

Drop mode argument to open(): ruff complains about open(mode="r")

aaronliu0130 commented on cpplint.py:
```diff
@@ -7590,16 +7590,15 @@ def ProcessFile(filename, vlevel, extra_check_functions=None):
-        # has CRLF endings.
+        # we are not opening the file with universal newline support,
+        # so the resulting lines do # contain trailing '\r' characters
+        # if we are reading a file that # has CRLF endings.
```
```suggestion
        # so the resulting lines do contain trailing '\r' characters
        # if we are reading a file that has CRLF endings.
```

aaronliu0130 replied:
Also, open() does have universal newlines support when newline=None. Should we adjust for that?

s-t-e-v-e-n-k replied:
I was wondering about that. I'm happy to circle back here on Monday and add in universal newlines support and see what kind of clean up that gives us.

s-t-e-v-e-n-k replied:
Done!

aaronliu0130 reviewed (changes requested):
I was planning to do this! Thank you.

s-t-e-v-e-n-k committed:
fix: Switch to using universal newlines

Since we now use plain open(), we can also use universal newlines and
remove large sections of code that keep track of CR and CRLF.
